### PR TITLE
Init Websockets

### LINKED ISSUE #273: [FEATURE] Emit Event via Websockets whenever a new Github Event is created
# ⭐ Suggestion

For the interactive Map project, the API needs to be able to establish WebSocket connections

# Concept

## Requirements
- API needs to be able to act as a WebSocket server
- Whenever a new Event from Github is tracked, we need to emit the event + the profile data via WebSockets to the connected clients
- following events need to be tracked
    - `join` when a new client connects
    - `getAllPointers` from retrieving all locations
    - `createPointer` for creating a new pointer on the map
    
## Implementation
- Usage of [Nestjs-Socket-io](https://docs.nestjs.com/websockets/gateways) Implementation
- As it is a mix of HTTP-Request and Websocket [this Codesandbox](https://codesandbox.io/s/floral-water-fbqey) might come handy

## Things to discuss
- How should we configure CORS - Should we wildcard it, or only allow specific URLs to connect?
- How do we want to document the WS-API? From my research[[1](https://github.com/nestjs/nest/issues/1794)] that's not supported with Swagger and also not planned

### PR DESCRIPTION
**Pull request by Cahllagerfeld**
closes #273

# Todo
- [x] populate created event with user

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/api/pull/276"></a>

### CODE REVIEW

**Comment by stemount** [comment]
@Cahllagerfeld Looks like this could just be merged straight away as it's just adding dependencies.

**Comment by Cahllagerfeld** [comment]
Its just a draft, (hopefully) there are a few things added to the PR

File: src/github/event.gateway.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+@WebSocketGateway({
+  cors: {
+    origin: '*',
+  },
+})
+export class EventGateway {
+  constructor(private readonly eventService: GithubEventService) {}
+
+  @SubscribeMessage('event')
```
Comment (Cahllagerfeld) [inline src/github/event.gateway.ts:18]: We may want to change the event-names

Comment (eddiejaoude) [reply src/github/event.gateway.ts:18]: maybe `github-events`?

**Review by eddiejaoude — APPROVED** [review]
Looks good 👍

**Pull request merged**